pcdscalc/be_lens_calcs.py

```diff
@@ -925,15 +954,428 @@ def find_z_pos(energy, lens_set, spot_size_fwhm, material=None,
     waist = lam / np.pi * focal_length / w_unfocused
     rayleigh_range = np.pi * waist ** 2 / lam
 
-    logger.info("waist          : %.3e" % waist)
-    logger.info("waist FWHM     : %.3e" % (waist
-                                           * FWHM_SIGMA_CONVERSION / 2.0))
-    logger.info("rayleigh_range : %.3e" % rayleigh_range)
-    logger.info("focal length   : %.3e" % focal_length)
+    logger.info("waist          : %.3e", waist)
+    logger.info("waist FWHM     : %.3e", waist * FWHM_SIGMA_CONVERSION / 2.0)
+    logger.info("rayleigh_range : %.3e", rayleigh_range)
+    logger.info("focal length   : %.3e", focal_length)
 
     w = gaussian_fwhm_to_sigma(spot_size_fwhm) * 2
     delta_z = rayleigh_range * np.sqrt((w / waist) ** 2 - 1)
     z1 = focal_length - delta_z
     z2 = focal_length + delta_z
     z_position = (z1, z2)
     return z_position
+
+
+def plan_set(energy, z_offset, z_range, beam_size_unfocused, size_horizontal,
+             size_vertical=None, exclude=None, max_tot_number_of_lenses=25,
+             max_each=5, focus_before_sample=False):
+    """
+    Macro to help plan for what lens set to use.
+
+    Parameters
+    ----------
+    energy : number
+        Beam energy in KeV
+    z_offset : float
+        Distance from sample to lens_z=0 in meters
+    z_range : list
+        Array or tuple of 2 values: minimum and maximum z pos in meters
+    beam_size_unfocused : float
+        Radial size of x-ray beam before focusing in meters
+    size_horiz
+    size_vert
```

Comment:
These two are missing parameter info and their names don't match the signature

Reply:
I honestly don't know what these two fields are... I was thinking that they could be the `Field of View horizontal`, `Field of View Vertical` - which are just numbers, in meters or millimeter  

But looking at the code, the `np.max` is taking `size_horizontal` as the `a` - array like, and the `size_vertical` as the `axis` - so this is confusing (unless they meant to use np.maximum??)
```
  if size_vertical is not None:
        size_calc_1 = np.max(size_horizontal, size_vertical)
    else:
        size_calc_1 = size_horizontal
```
That's why I had it empty for now until I get more feedback from the scientists.... unless you guys know what they are supposed to be?

Reply:
Fair enough! Thanks for looking into the types